openhtf/core/__init__.py

```diff
@@ -114,6 +114,10 @@ def _thread_proc(self):
       # Any access to self._exit_stack must be done while holding this lock.
       with self._lock:
         self._exit_stack = exit_stack
+        # We need to exit the PhaseExecutor before we tear down the plugs
+        # that a phase may be executing, so we add them in reverse order.
+        exit_stack.callback(self.test_state.plug_manager.tear_down_plugs)
+        exit_stack.callback(executor.stop)
```

Comment:
hm, by doing this here, the `PlugManager` may have `tear_down_plugs()` called before `initialize_plugs()`.  are you trying to address that `executor.stop()` is never called if the start trigger is broken?  can you not add `executor.stop()` to the block below (line 137 in the old version)?